osquery/config/config.cpp

```diff
@@ -105,6 +147,13 @@ Status Config::genConfig(OsqueryConfig& conf) {
         conf.options[v.first.data()] = v.second.data();
       }
     }
+
+    auto dispatch = osquery::Dispatcher::getInstance();
+    std::promise<EventFileMap_t> promise = std::promise<EventFileMap_t>(
+        std::allocator_arg, std::allocator<EventFileMap_t>());
+    conf.event_files = promise.get_future();
```

Comment:
This is a bit complicated. @marpaia maybe we should move the life of the Config to the life of the process. Then we can reduce this to a runnable future and a mutex lock. 

We'd also enable `Dispatcher::addService` for things like stateful config update monitors (via TCP/socket/etc). Of course that more-complicated config watching would be in plugins. 

For the filesystem config we'd lose the ability to "refresh" read the config, but it could support a os-specific file watcher to read in updates and apply them with a SIGHUP.